corehq/apps/hqadmin/views/users.py

```diff
@@ -158,24 +158,22 @@ def _validate_user_access(self, user):
         return True
 
     def get(self, request, *args, **kwargs):
-        self.is_web_user = request.GET.get('webuser', False)
+        self.domain = request.GET.get('domain')
         full_username = request.GET.get('as', '')
-        if not self.is_web_user:
-            if not full_username or '@' not in full_username:
-                return HttpResponseBadRequest('Please specify a user using ?as=user@domain')
 
+        if not full_username or '@' not in full_username:
+            msg = 'Please specify a user using ?as=user@domain\nOr specify a web-user using ?as=email&domain=domain'
+            return HttpResponseBadRequest(msg)
+
+        self.user = CouchUser.get_by_username(full_username)
+        if self.user.is_web_user and not self.domain:
+            msg = 'Please specify domain for web-user using ?as=email&domain=domain'
+            return HttpResponseBadRequest(msg)
+
+        if not self.user.is_web_user or not self.user:
```

Comment:
If the user is not found, this line will fail - you need to reverse the order of these two to guard against that.  Also, if the user provides the username in a normal format (as described [in the docs](https://confluence.dimagi.com/display/GS/Restore+File+and+Configuration+Options)), then `CouchUser.get_by_username` should've already found it.

Perhaps the answer here would be to move the `full_username` manipulation stuff higher up, something like:
- if there isn't an `@` sign, fail with a message
- if there's no `.` in the stuff after the `@` sign, assume it's `username@domain` and do the `format_username` stuff
- otherwise, leave it be.  This'll account for web users with normal email addresses, and mobile workers specified as `username@domain.commcarehq.org`

Then you can lookup the username just the once.